tests/unit/test_bugjira.py

```diff
@@ -0,0 +1,180 @@
+    WHEN we call it with no parameters
+    THEN a BrokerInitException should be raised
+    """
+    with pytest.raises(broker.BrokerInitException):
+        Bugjira()
+
+
+def test_init_with_only_bz_backend_supplied():
+    """
+    GIVEN the Bugjira class' constructor
+    WHEN we call it with only the bugzilla parameter supplied
+    THEN a BrokerInitException should be raised
+    """
+    with pytest.raises(broker.BrokerInitException):
+        Bugjira(bugzilla=Mock())
+
+
+def test_init_with_only_jira_backend_supplied():
+    """
+    GIVEN the Bugjira class' constructor
+    WHEN we call it with only the jira parameter supplied
+    THEN a BrokerInitException should be raised
+    """
+    with pytest.raises(broker.BrokerInitException):
+        Bugjira(jira=Mock())
+
+
+def test_get_broker_with_bugzilla_id(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance's _get_broker method
+    WHEN we call it with a key string that represents a bugzilla bug id
+    THEN the instance's _bugzilla_broker attribute should be returned
+    """
+    broker = sandboxed_bugjira._get_broker("123456")
+    assert broker == sandboxed_bugjira._bugzilla_broker
+
+
+def test_get_broker_with_jira_issue_id(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance's _get_broker method
+    WHEN we call it with a key string that represents a jira issue key
+    THEN the instance's _jira_broker attribute should be returned
+    """
+    broker = sandboxed_bugjira._get_broker("FOO-123")
+    assert broker == sandboxed_bugjira._jira_broker
+
+
+def test_get_issue_with_non_string_key(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance's get_issue method
+    WHEN we call it with a non-string parameter
+    THEN a ValueError is raised
+    """
+    with pytest.raises(ValueError):
+        sandboxed_bugjira.get_issue(1)
+
+
+def test_get_issue_good_bugzilla(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we invoke the get_issue method with a BZ number
+    THEN it should return an instance of Issue with a non-None bugzilla
+        attribute and a key matching the input
+    AND the instance's bugzilla attribute should have had its getbug method
+        invoked once
+    """
+    key = "123456"
+    issue = sandboxed_bugjira.get_issue(key)
+    assert isinstance(issue, BugzillaIssue)
+    assert issue.bugzilla
+    assert not issue.jira_issue
+    assert issue.key == key
+    assert sandboxed_bugjira.bugzilla.getbug.call_count == 1
+
+
+def test_get_issue_bad_bugzilla(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we invoke the get_issue method with a BZ number using a bug id that is
+        known not to correspond to any bug on the remote bugzilla service
+    THEN a Fault exception should be raised
+    """
+    sandboxed_bugjira.bugzilla.getbug.side_effect = Fault(
+        "Fault 101", "Bug #XXX does not exist."
+    )
+    with pytest.raises(Fault):
+        sandboxed_bugjira.get_issue("123")
```

Comment:
Should we wrap these errors in the get_issue method with our own, so a caller only has to catch that and not know all possible exceptions the underlying libraries might throw?

Reply:
+1, I think we should make it so the caller doesn't have to know the underlying libraries we use... I've been hesitating to suggest the same for the validation errors if possible, if they come directly from pydantic. Just wrap them into a ValidationError of our own, so that the caller doesn't have to do what libraries we use, and isn't affected if we change them.

Reply:
Good update, thanks!